Jenkinsfile.pull-request

```diff
@@ -0,0 +1,101 @@
+os = 'jammy'
+arch = 'amd64'
+flavor = 'server'
+type = 'DEB'
+def utils
+
+pipeline {
+  agent {
+    label 'linux && amd64'
+  }
+
+  options {
+    timestamps()
+    disableConcurrentBuilds()
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+  }
+
+  environment {
+        GITHUB_LOGIN = credentials('github-rstudio-jenkins')
+  }
+
+  stages {
+    stage('Load Utils') {
+      steps {
+        script {
+          sh 'printenv'
+          sh "echo 'Loading utils from ${env.WORKSPACE}/utils.groovy'"
+          utils = load "${env.WORKSPACE}/utils.groovy"
+          utils.addRemoteRef("${env.CHANGE_TARGET}")
+        }
+      }
+    }
+
+    stage('Build') {
+      when {
+        expression {
+          return utils.hasChangesIn('src') ||
```

Comment:
I think we need to include `dependencies/common`, `dependencies/tools`, and `dependencies/linux` here as well, as they are used in our build docker images. Also, `CMakeRelease.txt`, `VERSION.in`, `SOURCE.in`, `COPYING`, and `NOTICE`. We should also recompile for changes in `version/`. 

I think this method of explicitly listing everything that's in the root directory leaves us open to missing needed compiles if someone adds a new file. Similarly, if someone added another folder that's part of the build, I think we'd miss a compile based on that. If we could instead specify "changes in any folder except docs/", we'd be a lot less like to have to come and modify this in the future.

Reply:
I think you're right, in this case it's probably better to be overly _inclusive_ of changes that trigger rather than not inclusive enough. That way, he change triggers will match our main pipeline, too.